### DISCUSSION TITLE
[Need Help] How to access to Postal database since it running as docker ?

### DISCUSSION THREAD

**Discussion by XXXLukieDukieXXX** [postalserver/postal#3162] [Help with using Postal]
CONTAINER ID   IMAGE                               COMMAND                  CREATED       STATUS       PORTS                      NAMES
987612f34dbb   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   7 weeks ago   Up 7 weeks                              postal-worker-1
ae4f9e3a8520   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   7 weeks ago   Up 7 weeks                              postal-smtp-1
cf5149a3efc9   ghcr.io/postalserver/postal:3.3.4   "/docker-entrypoint.…"   7 weeks ago   Up 7 weeks                              postal-web-1
f95c7eff1a84   mariadb                             "docker-entrypoint.s…"   7 weeks ago   Up 7 weeks   127.0.0.1:3306->3306/tcp   postal-mariadb

---------------------------------------------------------------------------------------------------------------------------------------------------
#List the configuration database as below:
cat /opt/postal/config/postal.yml

main_db:
  host: 127.0.0.1
  username: root
  password: postal
  database: postal

message_db:
  host: 127.0.0.1
  username: root
  password: postal
  prefix: postal

How to access the db from docker? I have tested below command not working at all.

docker exec -it postal mysql -u root -p
docker exec -it  postal-mariadb mysql -u postal -p

**Comment by willpower232** [comment]
Did you already set up the postal user or are you trying to set it up now?

**Reply by XXXLukieDukieXXX** [reply]
I already setup the postal. Just trying to access db to check on the details. I managed to access to db. Thanks Willpower232

**Comment by UEAB-BL** [comment]
I just accessed mine using these commands:
`docker exec -it postal-mariadb sh`
`mariadb --user=user_name --password=your_password db_name`

**Reply by XXXLukieDukieXXX** [reply]
Got it. Thanks I able to accessing it. Appreciate your help.

**Reply by bukunmilab** [reply]
> I just accessed mine using these commands: `docker exec -it postal-mariadb sh` `mariadb --user=user_name --password=your_password db_name`

I think this is the best way to access MariaDB database of Postal via SSH. I just accessed mine using the same command.
For anyone who is going to use this, please ensure you change the username, password, and db_name to your actual credentials. You can see the credentials in your /opt/postal/config/postal.yml

@XXXLukieDukieXXX do you have some of MariaDB commands for Postal to tune the sending speed via SSH?
@willpower232 Please can you share any reference?

**Reply by UEAB-BL** [reply]
Also, you dont really need to type in the db name at all, if you ommit it you will still be connected, and then its just regular SQL commands, "show databases;" will let you know what its called. "postal" will probably be what you want to use =)

About tuning the sending speed, I just learned you can increase the amount of worker processes by using "postal start --scale worker=4". Thats not an SQL command of course though, you just run it on the server. You can ofc use another number than 4, default is 1.

**Comment by bukunmilab** [comment]
I am following this too. There are just a bunch of files named with my.cnf or mariadb.cnf or similar database configuration files and alternatives when you try to run `find / -name my.cnf` via SSH.
